pallets/author-slot-filter/src/lib.rs

```diff
@@ -128,54 +131,63 @@ pub mod pallet {
 
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
-		/// Update the eligible ratio. Intended to be called by governance.
+		/// Update the eligible count. Intended to be called by governance.
 		#[pallet::weight(T::WeightInfo::set_eligible())]
-		pub fn set_eligible(origin: OriginFor<T>, new: Percent) -> DispatchResultWithPostInfo {
+		pub fn set_eligible(
+			origin: OriginFor<T>,
+			new: EligibilityValue,
+		) -> DispatchResultWithPostInfo {
 			ensure_root(origin)?;
-			EligibleRatio::<T>::put(&new);
+			EligibleCount::<T>::put(&new);
 			<Pallet<T>>::deposit_event(Event::EligibleUpdated(new));
 
 			Ok(Default::default())
 		}
 	}
 
-	/// The percentage of active authors that will be eligible at each height.
+	/// The type of eligibility to use
+	pub type EligibilityValue = u32;
+
+	/// The number of active authors that will be eligible at each height.
 	#[pallet::storage]
-	#[pallet::getter(fn eligible_ratio)]
-	pub type EligibleRatio<T: Config> = StorageValue<_, Percent, ValueQuery, Half<T>>;
+	#[pallet::getter(fn eligible_count)]
+	pub type EligibleCount<T: Config> = StorageValue<_, EligibilityValue, ValueQuery, Half<T>>;
```

Comment:
This implies that a migration is necessary, it must also be decided how the percentage is translated into number during the migration, @alan do you have an idea on this?

Reply:
Since this might affect block production I suggest we keep the previous storage, and add the new EligibleCount in a new storage. We can in the future remove _EligibleRatio_

Reply:
Why do we want to keep the previous storage? How is it helping ?

Reply:
In general I am hesitant of migrating a storage prefix involved in block production without keeping the previous storage. I think that has been our approach so far with every data migration we have performed if we suspected it could affect block production

Reply:
Ok, we could mark it as deprecated